archive/100-theorems-list/37_solution_of_cubic.lean

```diff
@@ -0,0 +1,186 @@
+/-
+Copyright (c) 2022 Jeoff Lee. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeoff Lee
+-/
+import tactic.basic
+import ring_theory.roots_of_unity
+import ring_theory.polynomial.cyclotomic.basic
+
+/-!
+# The Solution of a Cubic
+
+This file proves Theorem 37 from the [100 Theorems List](https://www.cs.ru.nl/~freek/100/).
+
+The theorems `cubic_eq_zero_iff` and `cubic_eq_zero_iff_of_p_eq_zero` gives the solutions to the 
+cubic equation `a * x^3 + b * x^2 + c * x + d = 0` over a field `K` that has a third primitive 
+root of unity and that has characteristic neither 2 nor 3. 
```

Comment:
Looking at `cubic_eq_zero_iff`, and specifically the requirements on `r` and `s`, I believe you should add to the list of required properties of this field, that "certain other quantities admit square and cube roots".